### ISSUE TITLE
App rejected due to "VK API".

### ISSUE THREAD

**Issue by Vishaliphone** [viki-org/VKVideoPlayer#64]
Hi there,
I am using this library in my iOS app,
But my app is rejected by apple with below log :

"We found that your app allows users to download music or video content without authorization from the relevant third-party sources. Specifically, this app uses the VK API, which has been found associated with downloading third-party content."

**Comment by mohpor** [comment]
This is very unsettling on so many levels, but if you don't actually download third party contents and consider this rejection a complete fault of VKVideoPlayer, I strongly suggest you use Apple's support center and write down a detailed explanation (probably with links to this repo) and they'll listen, believe me, I have done this before (not related to VK API though).
Hope this helps

**Comment by Vishaliphone** [comment]
I had done what you mentioned above,
Below is a response from apple.

**Thank you for your correspondence. Moving forward, it would be appropriate to remove the VK video player from the app, as it has been found associated with downloading third-party content. 

We look forward to reviewing your revised app upon resubmission. 

Best Regards,
App Store Review**

**Comment by mohpor** [comment]
Then I guess we all have to follow suit! That's terrible news. I'm juts happy I didn't invest too much into it. Let me know if you've found something similar to this library. It was a great library and I wish I didn't have to move away from it.

**Comment by jonong** [comment]
Thanks for the feedback, Viki will be looking into this. That being said, there shouldn't be any code that downloads third party content unless the developer specifically adds in this functionality. We will contact Apple to see if we can figure out this issue. In the mean time, hang in there as we prepare a Swift update coming soon!

**Comment by mohpor** [comment]
We would really appreciate it if you do, and hey, it's an open source community, let us know if we can lend a hand and speed up the process.

That was good news.
Cheers.
M.

**Comment by mohpor** [comment]
@jonong  Could you give us a rough time frame for the upcoming swift framework?

**Comment by a13xb** [comment]
@Vishaliphone Out of curiosity, what's the app category, what does the app do?

**Comment by Vishaliphone** [comment]
@a13xb Take a look at my app 
[](https://itunes.apple.com/in/app/ilax-video-player-video-streamer/id539916247?mt=8)

It only manage Local video player & YouTube video player.
So, Basically app is all about Document manager with Local & YouTube video Play List manager.

**Comment by jonong** [comment]
@mohpor We're working on it right now, and should be coming in 2-3 weeks time. Stay tuned!

**Comment by mohpor** [comment]
@jonong  Awesome!

**Comment by sahin** [comment]
I suggest everyone to check out https://github.com/mobileplayer/mobileplayer-ios

**Comment by mohpor** [comment]
@sahin Not cool man :-1:

**Comment by sahin** [comment]
@mohpor :) it is cool to make people aware of new players. :)

**Comment by sahin** [comment]
@mohpor actually , you suggest this feature :)

**Comment by mohpor** [comment]
@sahin Of course, but you've just spammed the whole watch list and I've got like a million emails from this repo and it was annoying as hell.

There's a code of conduct for this kind of situations and you can rely on the ability of developers to come and find your framework by their own will, not by force feeding the news.

Good luck anyway.

**Comment by sahin** [comment]
you can rely on the ability of developers, that is for sure, I wish it is like that. then there will no marketing.. :) 
 
you make people annoyed that is for sure. I hated too, but we meet with  you you which is a good thing :) and it created some activity here, so they might fix some bugs in this repo too :)

**Comment by mohpor** [comment]
Marketing? I never knew open source projects needed marketing!

I'm not saying you shouldn't have said anything, but posting on one or two issues should've sufficed! 
And yet another reminder, they ARE working on a new version. Read this very thread to see.

**Comment by sahin** [comment]
@mohpor trying to making an open source project get more eyeballs sort of called marketing. 

![kenan-imirzalioglu-karadayi-dizisi-biyiklari](https://cloud.githubusercontent.com/assets/2508980/13068621/0c6e85b2-d432-11e5-890a-6e4fdf061d06.jpg)

**Comment by mohpor** [comment]
@sahin tamam

**Comment by sahin** [comment]
@mohpor tamam too :)

btw:
https://github.com/viki-org/VKVideoPlayer/issues/63#issuecomment-184547517

**Comment by jonong** [comment]
Hi @Vishaliphone, can you please pm me your app identifier so I can liaise with our Apple contacts please? We want to get down to why the app is being rejected, or what this "VK API" is, so that we can fix any potential issues with the library. Thanks!

**Comment by Vishaliphone** [comment]
Hi @jonong ,
Below is my App identifier: 
507942038

Thanks :)

**Comment by FridayDevGroup** [comment]
Excited for the swift version, thanks for the hard work.

**Comment by ThiagoMiranda** [comment]
Did you solve the rejecting issue?

**Comment by Vishaliphone** [comment]
Not yet.

**Comment by ThiagoMiranda** [comment]
@Vishaliphone Do you have a prediction date for that?
I'll begin to develop my company's video player SDK in IOS and I was willing to use yours to start from. The @sahin player has little support from his developers and it's very limited compared to VK one.

**Comment by vinoth56060** [comment]
Hi, Any News on App Rejection from App Store? Shall i go forward with this SDK?

**Comment by Vishaliphone** [comment]
Hallo all,
After get this issue, I did removed **VKVideoPlayer** from my project & implemented different player.
Though i did change my VideoPlayer, my app was rejected again with same issue.

So i contact Apple regarding this. They suggest me that there is [YTVimeoExtractor](https://github.com/lilfaf/YTVimeoExtractor) added in my project.So, i removed it & now my app is live.

So, i think you guys can continue with this awasome library.

Thanks!

**Comment by ThiagoMiranda** [comment]
@Vishaliphone which player framework did you use?

**Comment by Vishaliphone** [comment]
Hi,
Now i am using [https://github.com/kolyvan/kxmovie](kxmovie)

**Comment by sahin** [comment]
check out https://github.com/mobileplayer/mobileplayer-ios

**Comment by a120691696** [comment]
I have the same issue,Apple ask me remove all the VK API.

Our team has re-reviewed your binary with your question in mind, we found this binary includes classes and methods belonging to the VK API:

MemoryHogging (VKUtility)
VKFoundation (NSObject)
VKFoundation (NSString)
VKFoundation (UIImage)
VKPlayer (AVPlayer)
VKFoundation_stripHtml_XMLParsee
VKPickerButton
VKScrubber
VKSlider
VKThemeManager
VKUtility
VKVideoPlayer
VKVideoPlayerCaption
VKVideoPlayerCaptionSRT
VKVideoPlayerExternalMonitor
VKVideoPlayerLayerView
VKVideoPlayerSettingsManager
VKVideoPlayerTrack
VKVideoPlayerView
VKView


The above classes are samples of the SDK that should be removed prior to resubmission. Please ensure this binary does not include any remnants of the VK APIs including those not listed above.

**Comment by muzix** [comment]
Our app also got rejected.

> Your app falls into a category of apps that are often used for illegal file sharing, which does not comply with the App Store Review Guidelines. Specifically, we found that your app contains VK Video features.

**Comment by sahin** [comment]
@muzix @a120691696 @Vishaliphone guys I suggest https://github.com/mobileplayer/mobileplayer-ios/ , it is pretty good. Our top developers built it. and we are fast to fix the things

**Comment by muzix** [comment]
Unfortunately, we need support for iOS 7. But mobileplayer required iOS 8.

**Comment by elburro1887** [comment]
So this makes this library is pretty much useless, which is a shame.

Is there any way the devs can get in touch with apple directly and clear this up?

**Comment by FridaySG** [comment]
Try appealing and linking apple to this issue thread.

**Comment by muzix** [comment]
Good news, i tried appealing and give them link to this thread, now my app was accepted 💃 
Thanks everyone 👍

**Comment by tobitech** [comment]
Please can someone give me a summary of all that this library entails, things that one could likely encounter. I really want to try out this VKVideo player and I don't want to get my app rejected neither do I want to go through the process of writing to Apple. Would @muzix 's appeal solve the rejection issues once and forever??? I really want to know in depth before using this library in my app.

**Comment by muzix** [comment]
It will take time for Apple Review team to resolve the confusing between this lib and VK SDK. 
But I think appealing to Apple is not so hard and only take 1~2 working days.
And there isn't any other issues with this lib. In fact, this is a great player 👍 
If you considering writing app for iOS 8 and higher,  [mobileplayer-ios](https://github.com/mobileplayer/mobileplayer-ios/) seem promissing

**Comment by sahin** [comment]
thx, @muzix , a lot of people are moving to [mobileplayer-ios](https://github.com/mobileplayer/mobileplayer-ios/).

**Comment by kaizen1** [comment]
mobileplayer-ios uses Creative Commons license == Big problem for commercial projects. No go for me.

**Comment by sahin** [comment]
@kaizen1 we will change the license for commercial usage. It will be free. Please fell free to use it.
https://mobileplayer.io/pricing/

**Comment by victorstewart** [comment]
is this library still being rejected by apple?
